src/query/service/src/interpreters/access/privilege_access.rs

```diff
@@ -618,13 +619,49 @@ impl AccessChecker for PrivilegeAccess {
                 self.validate_db_access(&plan.catalog, &plan.database, UserPrivilegeType::Drop, plan.if_exists).await?;
             }
             Plan::UndropDatabase(_)
-            | Plan::DropUDF(_)
             | Plan::DropIndex(_)
             | Plan::DropTableIndex(_) => {
                 // undroptable/db need convert name to id. But because of drop, can not find the id. Upgrade Object to Database.
                 self.validate_access(&GrantObject::Global, UserPrivilegeType::Drop)
                     .await?;
             }
+            Plan::DropUDF(plan) => {
+                let udf_name = &plan.udf;
+                if !UserApiProvider::instance().exists_udf(&tenant, udf_name).await? && plan.if_exists {
+                    return Ok(());
+                }
+                if enable_experimental_rbac_check {
+                    let udf = HashSet::from([udf_name]);
+                    self.validate_udf_access(udf).await?;
+                } else {
+                    self.validate_access(&GrantObject::Global, UserPrivilegeType::Drop)
+                                        .await?;
+                }
+            }
+            Plan::DropStage(plan) => {
+                match UserApiProvider::instance().get_stage(&tenant, &plan.name).await {
```

Comment:
The user stage or tempory stage can get by this API? cc @flaneur2020 @youngsofun

Reply:
if i understands correctly, the user stage is not allowed to be dropped

Reply:
```
root@localhost:8000/default> drop stage ~;
error: APIError: ResponseError with 2506: user stage is not allowed to be dropped

```

Reply:
> The user stage or tempory stage can get by this API? cc @flaneur2020 @youngsofun

yes, I think

Reply:
what is `tempory stage`?

Reply:
> what is `tempory stage`?

is_temporary = true;

pub struct StageInfo {
    pub stage_name: String,
    pub stage_type: StageType,
    pub stage_params: StageParams,
    // on `COPY INTO xx FROM 's3://xxx?ak=?&sk=?'`, the URL(ExternalLocation) will be treated as an temporary stage.
    pub is_temporary: bool,

Reply:
@TCeason  no, `tempory stage` are not stored in meta server, and not accessible outside .

Reply:
> what is `tempory stage`?

is_temporary = true;

pub struct StageInfo {
    pub stage_name: String,
    pub stage_type: StageType,
    pub stage_params: StageParams,
    // on `COPY INTO xx FROM 's3://xxx?ak=?&sk=?'`, the URL(ExternalLocation) will be treated as an temporary stage.
    pub is_temporary: bool,